src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonPropertyInfoCharNullable.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonPropertyInfoCharNullable : JsonPropertyInfo<char?>, IJsonValueConverter<char?>
+    {
+        public JsonPropertyInfoCharNullable(Type classType, Type propertyType, PropertyInfo propertyInfo, JsonSerializerOptions options) :
+            base(classType, propertyType, propertyInfo, options)
+        { }
+
+        public bool TryRead(Type valueType, ref Utf8JsonReader reader, out char? value)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                value = default;
+                return false;
+            }
+
+            value = reader.GetString()[0];
+            return true;
+        }
+
+        public void Write(char? value, ref Utf8JsonWriter writer)
+        {
+#if BUILDING_INBOX_LIBRARY
+            char tempChar = value.Value;
+            Span<char> tempSpan = MemoryMarshal.CreateSpan<char>(ref tempChar, 1);
+            writer.WriteStringValue(tempSpan);
+#else
+            writer.WriteStringValue(value.ToString());
+#endif
```

Comment:
And in a case like this, I doubt CreateSpan is really buying you much over just doing a stackalloc char[1] (I could be wrong of course), so you could just use that code in all builds, e.g.
```C#
Span<char> tempSpan = stackalloc char[1] { value.Value };
writer.WriteStringValue(tempSpan);
```